### DISCUSSION TITLE
Coolify Could Not Connect to the New Realtime Service

### DISCUSSION THREAD

**Discussion by darrylmorley** [coollabsio/coolify#1682] [Q&A]
Hi,
I'm seeing the error `Coolify could not connect to the new realtime service introduced in beta.154.
This will cause unusual problems on the UI if not fixed!`. Does anyone know what this is about, maybe how to resolve?

This is a fresh install on Ubuntu 22.04, AWS EC2.

**Comment by pfallasro** [comment]
I was having the same issue, and then I noticed I was missing one of the ports they ask for in the documentation.

**Reply by YashwanthCH18** [reply]
which port is that??

**Reply by ardabeyazoglu** [reply]
6001

**Comment by Matze202** [comment]
I have the same problem, can someone please help and provide a solution or post a link that will help?

**Comment by ardabeyazoglu** [comment]
If anyone is experiencing this issue with nginx as a reverse proxy, websockets ports need to be proxied as well as the core application.

Example configuration with recommended installation, and a custom nginx-proxy-manager:
![image](https://github.com/user-attachments/assets/00320de1-14fe-4b62-bd51-22640176ae1d)

**Reply by douglasg14b** [reply]
This is a lifesaver, docs need this info.

Here is a caddyfile config for the same:

```yml

coolify.yourdomain.com {
        @terminal {
                path /terminal/ws /terminal/ws/*
        }

        @app {
                path /app/*
        }

        handle @terminal {
                reverse_proxy 192.168.30.9:6002
        }

        handle @app {
                reverse_proxy 192.168.30.9:6001
        }

        reverse_proxy 192.168.30.9:8000
}
```

**Reply by ardabeyazoglu** [reply]
Nice!

**Reply by Snuffsis** [reply]
Which version of nginx-proxy-manager is this?
I am running 2.11.1 and I have the issue that, when I am adding the locations, the proxy host ends up in "offline" status and won't work.
Did you set it up on the proxy host for the coolify itself?

**Reply by AmrouHab** [reply]
> Which version of nginx-proxy-manager is this? I am running 2.11.1 and I have the issue that, when I am adding the locations, the proxy host ends up in "offline" status and won't work. Did you set it up on the proxy host for the coolify itself?

Having the same issue with NPM vers 2.10.4, did you find any solution ? 

Edit : Updated to 2.12.3 and it works, be sure to use IP instead of hostname, using hostnames causes the issue

**Reply by rjarman** [reply]
This setup is working for me in Cloudflare. I was getting broken requests for real-time WS calls. 
![Screenshot from 2025-06-04 18-40-27](https://github.com/user-attachments/assets/ca763da7-dbee-4ea9-8c6e-03ac7c9f31df)

**Reply by djsisson** [reply]
if you have entered an instance domain for coolify, then you can replace the top 3 lines with just the same entry as your bottom ones ```localhost 80```

and since it looks like you have a wildcard for your last entry, then you do not need to add the coolify entrys at all

**Reply by rjarman** [reply]
Did not work for me, rather I have to add another entry for build/assets path(1st entry). 
In the cloudflare logs, it's showing that the https://coolify.customdomain.com/build/assets/* are redirecting to 6001. So I have added another one like this otherwise assets are getting 404:
```
hostname: coolify.domain.com
path: build/assets
service: http://localhost:8000
```

**Reply by BayPhillips** [reply]
@rjarman I'm trying to do the above, but when I match what you have I cannot load any of the project pages and am just getting 404s (the rest of the dashboard works fine). If I move the app path to below the * match it loads correctly again so my guess is it's matching the path the path as it has *app*lication in it 
`/project/${projectId}/environment/${environmentId}/application/${applicationId}`. Did you ever run into this?

**Reply by rjarman** [reply]
@BayPhillips currently I am using these settings, and they are working fine without entering the asset paths

![Public hostnames - Cloudflare One](https://github.com/user-attachments/assets/94a01a49-1d17-4ef2-a7f2-14a60359cc7c)

**Reply by DominusKelvin** [reply]
> This is a lifesaver, docs need this info.
> 
> Here is a caddyfile config for the same:
> 
> ```yaml
> coolify.yourdomain.com {
>         @terminal {
>                 path /terminal/ws /terminal/ws/*
>         }
> 
>         @app {
>                 path /app/*
>         }
> 
>         handle @terminal {
>                 reverse_proxy 192.168.30.9:6002
>         }
> 
>         handle @app {
>                 reverse_proxy 192.168.30.9:6001
>         }
> 
>         reverse_proxy 192.168.30.9:8000
> }
> ```

This worked for me. Why isn't it in the docs @andrasbacsai?

**Reply by djsisson** [reply]
coolify does generate this when you set a domain for coolify, i assume its because once up and running you would only have 80/443 open and all other ports blocked

**Comment by bendtherules** [comment]
For cloudflare tunnel -
Previous comments suggest simple paths like /app, but these can be brittle or cause "path bleed" where they accidentally intercept dashboard routes (like /project/.../application/...).

Use this proper regex - `^/app($|/.*)` , `^/terminal/ws($|/.*)` (both above `*` route)

**Comment by aboqasem** [comment]
Using Cloudflare Tunnels:

<img width="819" height="180" alt="image" src="https://github.com/user-attachments/assets/f09e386a-6045-4e05-8eba-ce6de9e81dba" />

**Reply by voronkov-oleksandr** [reply]
thanks mate, forgot about this regex feature of cf tunnels
